### PR TITLE
cbor: Bounds checking on read and no_copy string deserialization

### PR DESCRIPTION
**Pull request by luciotorre**
Added bounds checking on deserialization of int, double, byte_array and unicode.
Added zero copy version of string deserialization.
With some tests.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@tfar, ok for you to be assigned?

**Comment by luciotorre** [comment]
Can you guys advise on what we can do to help land this?

**Comment by Kijewski** [comment]
Summoning @krf.

**Comment by krf** [comment]
Could you split out the uncontroverisal patches, i.e. `dont double define _XOPEN_SOURCE` and rebase+squash this one?

**Comment by krf** [comment]
@luciotorre: Different question: Are you using the CBOR module in the RIOT context or for another project? (I'm still wondering whether it's worth providing a standalone version of this implementation here, i.e. in a separate repository)

**Comment by luciotorre** [comment]
I am using it standalone. It would be great to have it separated.

**Comment by krf** [comment]
@luciotorre: Cool! Feel free to reach out to me at kfunk@kde.org, so we can set that up. I'm interested if you have other patches that are needed to make this work.

Note: Still happy to review the code *here*, though. (Because there's CI coverage, and we have the valuable input of the RIOT guys)

**Comment by krf** [comment]
@luciotorre: I've started https://github.com/krf/cbor -- still need to polish (finish README, etc.)

**Comment by OlegHahm** [comment]
@krf, did you find time to review?

**Comment by krf** [comment]
@OlegHahm: I did, see https://github.com/RIOT-OS/RIOT/pull/3843#commitcomment-16067063 and following

**Comment by OlegHahm** [comment]
Oops, sorry!

**Comment by OlegHahm** [comment]
@luciotorre, ping!

**Comment by OlegHahm** [comment]
Sorry, feature freeze.

**Comment by luciotorre** [comment]
Hey guys, sorry, release time over here. Ill look at this when the
crazyness goes away.

File: core/include/byteorder.h [hunk 1]
```diff
@@ -21,8 +21,14 @@
...
-#if defined(__MACH__)
-#   include "clang_compat.h"
+#if defined(__clang__)
+# undef HTONS
+# undef HTONL
+# undef HTONLL
+# undef NTOHS
+# undef NTOHL
+# undef NTOHLL
+# undef HAVE_NO_BUILTIN_BSWAP16
```
Comment (miri64) [inline core/include/byteorder.h]: What do those changes have to do with CBOR?

**Comment by kYc0o** [comment]
Needs rebase. Let's look into this one for the next release.

**Comment by miri64** [comment]
How is it going here? A rebase would be nice :-)

**Comment by OlegHahm** [comment]
Let's continue in #5203.

**Pull request closed without merging**